src/AggregateFunctions/AggregateFunctionSparkbar.h

```diff
@@ -43,7 +43,16 @@ struct AggregateFunctionSparkbarData
 
         auto [it, inserted] = points.insert({x, y});
         if (!inserted)
-            it->getMapped() += y;
+        {
+            if (std::numeric_limits<Y>::max() - it->getMapped() > y)
```

Comment:
In fact, it is ok to annotate it with `NO_SANITIZE_UNDEFINED`, because the signed integer overflow is UB from the C++ compiler's standpoint but not from the CPU standpoint. We use it in other arithmetic operations, such as `plus`

Reply:
@alexey-milovidov  Unsigned overflow is also possible. In such case, saturating is a more expected result than losing values with passing through zero, imho. The `arithmeticOverflow.h` allows only overflow. Should we introduce functions for saturating arithmetic?
Another approach is to use UInt64 as a counter type for all smaller types (and handle only  UInt64 overflow).

Reply:
Switched to `common:addOverflow` with checking the result and saturating